tools/cli/commands/build.js

```diff
@@ -768,6 +770,35 @@ async function buildProject( t ) {
 		if ( composerJson.repositories.length === 0 ) {
 			delete composerJson.repositories;
 		}
+
+		// Update dependency version numbers in composer.json.
+		const composerDepTyes = [ 'require', 'require-dev' ];
+		for ( const key of composerDepTyes ) {
+			if ( composerJson[ key ] ) {
+				for ( const [ pkg ] of Object.entries( composerJson[ key ] ) ) {
+					for ( const ctxPkg of Object.values( t.ctx.versions ) ) {
+						if ( ctxPkg.name === pkg ) {
```

Comment:
```suggestion
						if ( ctxPkg.name === pkg && ctxPkg.version.includes( '@dev' ) ) {
```

> ...I wonder if there should be a test for the dep being @dev so it doesn't overwrite the non-@dev version numbers in release branches.

Reply:
Should work

Reply:
Strike that, it won't work. You need to be checking the version from composerJson, not the one in `ctxPkg`.

Reply:
5de057b